File: src/main/java/com/fish/extendedae_plus_client/integration/ContextModLoaded.java
package com.fish.extendedae_plus_client.integration;

import net.neoforged.fml.ModList;

public enum ContextModLoaded {
    emi("emi"),
    jei("jei"),
    jech("jecharacters"),
    curios("curios"),
    ae2wtlib("ae2wtlib"),
    advancedAE("advanced_ae"),
    appliedFlux("appflux"),
    mekanism("mekanism"),
    appliedMekanistics("appmek"),
    gtceuModern("gtceu"),

    ;

    private static boolean initialized;

    private final String modID;
    private boolean loaded;

    ContextModLoaded(String modID) {
        this.modID = modID;
        this.loaded = false;
    }

    public boolean isLoaded() {
        return this.loaded;
    }

    public static void init() {
        if (initialized)
            throw new IllegalStateException("Contexts has already been initialized");
        initialized = true;

        for (ContextModLoaded context : ContextModLoaded.values()) {
            context.loaded = ModList.get().isLoaded(context.modID);
        }
    }
}

File: src/main/kotlin/com/fish/extendedae_plus_client/integration/ContextModLoaded.kt
package com.fish.extendedae_plus_client.integration

import net.neoforged.fml.ModList

@Suppress("EnumEntryName")
enum class ContextModLoaded(private val modID: String) {
    emi("emi"),
    jei("jei"),
    jech("jecharacters"),
    curios("curios"),
    ae2wtlib("ae2wtlib"),
    advancedAE("advanced_ae"),
    appliedFlux("appflux"),
    mekanism("mekanism"),
    appliedMekanistics("appmek"),
    gtceuModern("gtceu"),
    ftbLibrary("ftblibrary"),
    ;

    var isLoaded: Boolean
        private set

    init {
        this.isLoaded = false
    }

    companion object {
        private var initialized = false

        fun init() {
            check(!initialized) { "Contexts has already been initialized" }
            initialized = true

            for (context in entries) {
                context.isLoaded = ModList.get().isLoaded(context.modID)
            }
        }
    }
}
